insurance/data_generation.ipynb

```diff
@@ -130,16 +145,30 @@
    "outputs": [],
    "source": [
     "class DBCONFIG:\n",
-    "    def __init__(self, hostname, databasename, username, password):\n",
+    "    def __init__(self, hostname, databasename, username, password, provider=\"mysql\"):\n",
     "        self.hostname = hostname\n",
     "        self.databasename = databasename\n",
     "        self.username = username\n",
     "        self.password = password\n",
+    "        if provider not in [\"mysql\", \"mssql\"]:\n",
+    "            raise ValueError(f\"{provider} not a supported provider\")\n",
+    "        self.provider = provider\n",
     "    \n",
     "    def __str__(self):\n",
-    "        return f\"mysql+pymysql://{self.username}:{self.password}@{self.hostname}/{self.databasename}\"\n",
+    "        if self.provider == \"mysql\":\n",
+    "            return f\"mysql+pymysql://{self.username}:{self.password}@{self.hostname.split(':')[0]}:{self.hostname.split(':')[1]}/{self.databasename}\"\n",
+    "        elif self.provider == \"mssql\":\n",
+    "            params =  quote_plus(\"DRIVER={ODBC Driver 17 for SQL Server};\"\n",
+    "                                 f\"SERVER={','.join(self.hostname.split(':'))};\"\n",
+    "                                 f\"DATABASE={self.databasename};\"\n",
+    "                                 f\"UID={self.username};\"\n",
+    "                                 f\"PWD={self.password}\")\n",
+    "            return f\"mssql+pyodbc:///?odbc_connect={params}\"\n",
+    "        else:\n",
+    "            return f\"DBCONFIG<{self.hostname}, {self.databasename}, {self.username}, {self.password}, {self.provider}>\"\n",
```

Comment:
What db provider does else  part cover? Raise exception otherwise.

Reply:
in __init__ it will raise exception already i can remove this code it will unreachable